### DISCUSSION TITLE
Improvement for `phpdoc_order` and/or `phpdoc_separation`

### DISCUSSION THREAD

**Discussion by DaedalusDev** [PHP-CS-Fixer/PHP-CS-Fixer#7238] [Feature ideas]
## Feature request

Is it possible :
- To "merge" `phpdoc_order` fixer with  `phpdoc_separation` fixer OR
- Add an option `order` to `phpdoc_separation` OR
- Add support for wildcard to `phpdoc_order` (same as https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/6635)

I think i can create submit some pull requests but i prefer ask what is the prefered approch for maintainers.

Best regards.

**Comment by Wirone** [comment]
Hi @DaedalusDev, really sorry for the late response. IMHO those 2 fixers could be merged, because their intent is pretty similar, but here are things that should be taken into consideration:

- default behavior of both should remain the same to prevent backward compatibility.
- personally I would introduce new `phpdoc_annotation_order` and deprecate both `phpdoc_order` and `phpdoc_separation`. Both of deprecated fixers should proxy calls to new fixer with proper configuration.
- new fixer's `order` config option should support both `['a', 'b', 'c', 'd']` and `[['a', 'b'], ['c', 'd']]` formats, simply adding empty line before every group

But this is only my perspective, we need other opinions (@kubawerlos, @keradus).
